js/apps/account-ui/src/applications/Applications.tsx

```diff
@@ -128,15 +128,22 @@ const Applications = () => {
                 className="pf-u-align-items-center"
                 dataListCells={[
                   <DataListCell width={2} key={`client${application.clientId}`}>
-                    <Button
-                      className="pf-u-pl-0 title-case"
-                      component="a"
-                      variant="link"
-                      onClick={() => window.open(application.effectiveUrl)}
-                    >
-                      {application.clientName || application.clientId}{" "}
-                      <ExternalLinkAltIcon />
-                    </Button>
+                    {application.effectiveUrl && (
+                      <Button
+                        className="pf-u-pl-0 title-case"
+                        component="a"
+                        variant="link"
+                        onClick={() => window.open(application.effectiveUrl)}
+                      >
+                        {application.clientName || application.clientId}{" "}
+                        <ExternalLinkAltIcon />
+                      </Button>
+                    )}
+                    {!application.effectiveUrl && (
```

Comment:
Rather than inverting the condition here, you could use a ternary:

```tsx
{application.effectiveUrl ? (<Button></Button>) : (<span></span>);
```

Reply:
I personally find that the ternary operator can make it hard to read, especially when large blocks of code are involved.